frontend/src/main/components/DriverAvailability/DriverAvailabilityTable.js

```diff
@@ -0,0 +1,63 @@
+import OurTable, {ButtonColumn} from "main/components/OurTable"  // if admin, needs {ButtonColumn}
+import { useBackendMutation } from "main/utils/useBackend";
+import { cellToAxiosParamsDelete, onDeleteSuccess } from "main/utils/driverAvailabilityUtils"
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function DriverAvailabilityTable ({ driverAvailabilities, currentUser }) { // if admin, needs currentUser
+    const navigate = useNavigate();
+
+    const editCallback = (cell) => {
+        navigate(`/driverAvailability/edit/${cell.row.values.id}`)
+    }
+    
+
+    // Stryker disable all : hard to test for query caching
+
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/driverAvailability/all"]
+    );
+    // Stryker restore all 
+
+    // Stryker disable next-line all : TODO try to make a good test for this
+    const deleteCallback = async (cell) => { deleteMutation.mutate(cell); }
+    const columns = [
+        {
+            Header: 'id',
+            accessor: 'id', // accessor is the "key" in the data
+        },
+        {
+            Header: 'Day',
+            accessor: 'day',
+        },
+        {
+            Header: 'Start Time',
+            accessor: 'startTime',
+        },
+        {
+            Header: 'End Time',
+            accessor: 'endTime',
+        },
+        {
+            Header: 'Driver ID',
+            accessor: 'driverId',
+        },
+        {
+            Header: 'Notes',
+            accessor: 'notes',
+        }
+    ];
+    if (hasRole(currentUser, "ROLE_ADMIN")) {
+        columns.push(ButtonColumn("Delete", "danger", deleteCallback, "DriverAvailabilityTable"));
+    } 
+    if (hasRole(currentUser, "ROLE_ADMIN") || hasRole(currentUser, "ROLE_DRIVER") ){
+        columns.push(ButtonColumn("Edit", "primary", editCallback, "DriverAvailabilityTable"));
```

Comment:
Mismatch between what the frontend shows and the backend capabilities.

Right now the backend only allows drivers to do anything with driveravailability. Im going to request changes to the backend, so the buttons should both show up for admins and drivers

Reply:
So both driver and admin should be able to edit and delete availability?

Reply:
<img width="1998" alt="image" src="https://github.com/ucsb-cs156-w24/proj-gauchoride-w24-7pm-4/assets/47500460/2e5a475f-6193-4f44-9456-bc8e05307e1c">